pandas/tests/copy_view/test_setitem.py

```diff
@@ -59,17 +60,25 @@ def test_set_column_with_index(using_copy_on_write):
     assert not np.shares_memory(get_array(df, "c"), idx.values)
 
     # and thus modifying the index does not modify the DataFrame
-    idx.values[0] = 0
-    tm.assert_series_equal(df["c"], Series([1, 2, 3], name="c"))
+    if using_copy_on_write:
+        with pytest.raises(ValueError, match="assignment"):
+            idx.values[0] = 0
```

Comment:
This defeats a bit the purpose of the original test (and I don't think we need to test here that assigning into a read-only numpy array gives an error). 
So maybe just remove it? (we already check `np.shares_memory`) Or manually set the writeable flag to True and then assign the value. 

But actually, now that we keep track of references to Index data as well, the original setitem doesn't really need to do a copy, I think? (for another issue/PR)

Reply:
Yeah let's rip it out